tests/genesis/genesis_test.go

```diff
@@ -0,0 +1,737 @@
+package genesis
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sort"
+	"testing"
+
+	"github.com/iancoleman/strcase"
+	"github.com/oasisprotocol/oasis-core/go/common/entity"
+	"github.com/oasisprotocol/oasis-core/go/common/node"
+	governance "github.com/oasisprotocol/oasis-core/go/governance/api"
+	registry "github.com/oasisprotocol/oasis-core/go/registry/api"
+	staking "github.com/oasisprotocol/oasis-core/go/staking/api"
+	oasisConfig "github.com/oasisprotocol/oasis-sdk/client-sdk/go/config"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/oasislabs/oasis-indexer/log"
+	"github.com/oasislabs/oasis-indexer/storage"
+	"github.com/oasislabs/oasis-indexer/storage/oasis"
+	"github.com/oasislabs/oasis-indexer/storage/postgres"
+	"github.com/oasislabs/oasis-indexer/tests"
+)
+
+type TestEntity struct {
```

Comment:
We are not able to re-use the API types?

Reply:
Technically we can. But we don't always want to validate all data, so I prefer having separate test types sometimes. And makes it easier to mock things when needed in a separate package rather than put it in the non-test one. But certainly a stylistic choice.